### PR TITLE
Only look at top-level private/public calls

### PR DESCRIPTION
**Pull request by jez**
### Motivation

Only consider sends at (or near) the top level of a class def, but
**not** inside any other definition (like a method).

This makes it easier to implement `private` with no arguments.

### Test plan

Check the commit history for a before/after version of the test.

### CODE REVIEW

File: test/testdata/rewriter/flatten_module_private_class_method.rb.symbol-table-raw.exp [hunk 1]
```diff
@@ -22,6 +20,4 @@ class <C <U <root>>> < <C <U Object>> ()
       argument <blk><block> @ Loc {file=test/testdata/rewriter/flatten_module_private_class_method.rb start=??? end=???}
     method <S <C <U B>> $1><U thing> (<blk>) @ Loc {file=test/testdata/rewriter/flatten_module_private_class_method.rb start=11:3 end=11:17}
       argument <blk><block> @ Loc {file=test/testdata/rewriter/flatten_module_private_class_method.rb start=??? end=???}
-  class <S <S <C <U B>> $1> $1>[<C <U <AttachedClass>>>] < <S <S <C <U Object>> $1> $1> () @ Loc {file=test/testdata/rewriter/flatten_module_private_class_method.rb start=10:7 end=10:8}
-    type-member(+) <S <S <C <U B>> $1> $1><C <U <AttachedClass>>> -> LambdaParam(<S <S <C <U B>> $1> $1><C <U <AttachedClass>>>, lower=T.noreturn, upper=AppliedType {   klass = <S <C <U B>> $1>   targs = [     <C <U <AttachedClass>>> = B   ] }) @ Loc {file=test/testdata/rewriter/flatten_module_private_class_method.rb start=10:7 end=10:8}
```
Comment (jez) [inline test/testdata/rewriter/flatten_module_private_class_method.rb.symbol-table-raw.exp:26]: Check the commit messages for a lengthy description of this file's diff.

File: namer/namer.cc [hunk 2]
```diff
@@ -467,6 +467,12 @@ class SymbolFinder {
     }
 
     ast::TreePtr postTransformSend(core::Context ctx, ast::TreePtr tree) {
+        // Only consider sends at (or near) the top level of a class def, but **not** inside any
+        // other definition (like a method).
+        if (ownerStack.back().kind() != DefinitionKind::Class) {
```
Comment (jvilk-stripe) [inline namer/namer.cc:472]: There's actually a way to do this that makes namer faster: Make this pass a `ShallowMap`!

Comment (jvilk-stripe) [reply namer/namer.cc:472]: (We might be able to do the same with `TreeSymbolizer`? Not sure.)

Comment (jvilk-stripe) [reply namer/namer.cc:472]: (Unless we otherwise care about definitions within method definitions? I don't know their semantics in Ruby.)

Comment (elliottt) [reply namer/namer.cc:472]: +1 to this, I like the idea of skipping method bodies unless we actually have to traverse them.

**Comment by jez** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_Ib2h2OVmMUrDlD
→ https://go/builds/bui_Ib2hgsemCisRHE

**Comment by jez** [comment]
ptal @elliottt 

Stripe codebase passes, test suite passes, and I can't imagine this makes sorbet slower (but i'm too lazy to measure)

**Comment by elliottt** [comment]
> ptal @elliottt
> 
> Stripe codebase passes, test suite passes, and I can't imagine this makes sorbet slower (but i'm too lazy to measure)

If anything, I think this would improve performance as we'd be avoiding method body traversals.

**Review by elliottt — APPROVED** [review]

**Pull request merged**